draft-normington-jsonpath-latest.md

```diff
@@ -425,22 +428,25 @@ An array index is a union element consisting of an integer (in base 10).
 ~~~~ abnf
 array-index = integer
 
-integer = [%x2D] (%x30 / (%x31-39 *%x30-39))
+integer = ["-"] ("0" / (%x31-39 *%x30-39))
                             ; optional - followed by 0 or
                             ; sequence of digits with no leading zero
+DIGIT1 = %x31-39            ; non-zero digit
 ~~~~
 
 Note: the syntax does not allow integers with leading zeros such as `01` and `-01`.
 
-An array slice is a union element consisting of two or three integers (in
-base 10 and which may be omitted) separated by colons.
+An array slice is a union element consisting of optional integers (in base 10) separated by colons.
```

Comment:
I think we need to be more aware about saying "is" when we mean "is represented by".

Reply:
Actually, I mean "is" rather than "is represented by". But the phrase "is a union element consisting of" is causing a lot of grief. It probably is better to make the array slice description independent of where it is used, thus:
```suggestion
An array slice consists of optional integers (in base 10) separated by colons.
```